### ISSUE TITLE
Job submissions can break functionality of "kubectl get pods"

### ISSUE THREAD

**Issue by jayunit100** [GoogleCloudPlatform/kubernetes#3461]
Submitting a incorrect yaml file for kubectl creation breaks the "get pods" command, somehow... This is obviously a bug since a user's submission error shouldn't result in an inability to view status of the existing pods.

*DETAILS* 

the following yaml

```
 cat spk-kube.yaml 
id: spk
kind: Pod
apiVersion: v1beta1
desiredState:
  manifest:
    version: v1beta1
    id: spark
    containers:
      - name: spk
        image: spk
        cpu: 1000
        ports:
          - name: sp
            containerPort: 7077
labels:
  name: spk
```

if launched w / `kube create -f ...` will cause `kubectl get pods` to report the following error message.

*get.go:75] The string provided is not a valid PodPhase constant value*

**Comment by jayunit100** [comment]
*workaround* after submitting a job which breaks this, you can delete the same pod.  This will rescue the behaviour  of `kubectl get pods`..

**Comment by jayunit100** [comment]
redis-master-pod.json <-- also i heard a report that this file can also be used to break `kubectl`

**Comment by jayunit100** [comment]
TL;DR  There are different ways that submitting a yaml or json file can break `kubectl get pods`.

**Comment by lavalamp** [comment]
Actually I think the problem is an old client. What version is your kubectl and what version is your cluster?

**Comment by jayunit100** [comment]
interesting.  maybe you are right , i just saw this.   i will sync versions and confirm if that rescues.  
```Client Version: version.Info{Major:"0", Minor:"7+", GitVersion:"v0.7.0-18-g52e165a4fd720d-dirty", GitCommit:"52e165a4fd720d1703ebc31bd6660e01334227b8", GitTreeState:"dirty"}
Server Version: &version.Info{Major:"0", Minor:"8+", GitVersion:"v0.8.0-248-gc93fb4e309aba8", GitCommit:"c93fb4e309aba8d8562dcd4c73cc162637278ac2", GitTreeState:"clean"}
```

**Comment by lavalamp** [comment]
Yeah, with that combo if for some reason the pod phase is "unknown" the .7 client will barf. More recent clients have a better error message for this case.

**Comment by bgrant0607** [comment]
Enums are not extensible

**Comment by jayunit100** [comment]
okay, closing this issue b/c its simply that cli/server not in sync
